File: proxy-server.js
// Servidor Proxy simples para contornar CORS
// Execute com: node proxy-server.js

const express = require('express');
const cors = require('cors');
const fetch = require('node-fetch');

const app = express();
const PORT = 3000;

// Habilitar CORS para todas as origens
app.use(cors());

// Rota do proxy
app.get('/api/*', async (req, res) => {
    try {
        // Extrair a URL real do parâmetro
        const targetUrl = req.query.url;

        if (!targetUrl) {
            return res.status(400).json({ error: 'URL não fornecida' });
        }

        console.log(`📡 Proxy request: ${targetUrl}`);

        // Fazer a requisição
        const response = await fetch(targetUrl);

        if (!response.ok) {
            console.error(`❌ Steam API error: ${response.status} ${response.statusText}`);
            const errorText = await response.text();
            console.error('Resposta:', errorText);
            return res.status(response.status).json({
                error: `Steam API error: ${response.status}`,
                details: errorText
            });
        }

        const data = await response.json();
        console.log('✅ Resposta recebida com sucesso');

        // Retornar os dados
        res.json(data);
    } catch (error) {
        console.error('❌ Erro no proxy:', error);
        res.status(500).json({ error: error.message });
    }
});

app.listen(PORT, () => {
    console.log(`🚀 Servidor proxy rodando em http://localhost:${PORT}`);
    console.log(`📝 Use: http://localhost:${PORT}/api/?url=SUA_URL_AQUI`);
});

File: proxy-server.py
# Servidor Proxy simples para contornar CORS
# Execute com: python proxy-server.py

from http.server import HTTPServer, BaseHTTPRequestHandler
from urllib.parse import urlparse, parse_qs, unquote
import urllib.request
import json

class ProxyHandler(BaseHTTPRequestHandler):
    def do_GET(self):
        # Parse da URL
        parsed_path = urlparse(self.path)
        
        if parsed_path.path == '/api/':
            # Extrair a URL target dos parâmetros
            params = parse_qs(parsed_path.query)
            
            if 'url' not in params:
                self.send_error(400, 'URL não fornecida')
                return
            
            target_url = unquote(params['url'][0])
            
            try:
                print(f'Proxy request: {target_url}')
                
                # Fazer a requisição
                with urllib.request.urlopen(target_url) as response:
                    data = response.read()
                
                # Enviar resposta com CORS headers
                self.send_response(200)
                self.send_header('Content-Type', 'application/json')
                self.send_header('Access-Control-Allow-Origin', '*')
                self.send_header('Access-Control-Allow-Methods', 'GET, OPTIONS')
                self.send_header('Access-Control-Allow-Headers', 'Content-Type')
                self.end_headers()
                self.wfile.write(data)
                
            except Exception as e:
                print(f'Erro no proxy: {e}')
                self.send_error(500, str(e))
        else:
            self.send_error(404, 'Rota não encontrada')
    
    def do_OPTIONS(self):
        # Handle preflight requests
        self.send_response(200)
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'GET, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')
        self.end_headers()

def run_server(port=3000):
    server_address = ('', port)
    httpd = HTTPServer(server_address, ProxyHandler)
    print(f'🚀 Servidor proxy rodando em http://localhost:{port}')
    print(f'📝 Use: http://localhost:{port}/api/?url=SUA_URL_AQUI')
    httpd.serve_forever()

if __name__ == '__main__':
    run_server()
